src/platform/posix/posix_pollq_kqueue.c

```diff
@@ -0,0 +1,392 @@
+#ifdef NNG_HAVE_KQUEUE
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h> /* for strerror() */
+#include <sys/event.h>
+#include <unistd.h>
+
+#include "core/nng_impl.h"
+#include "platform/posix/posix_pollq.h"
+
+#if defined NNG_PLATFORM_NETBSD
```

Comment:
It would be nice to have this detected by a check, rather than using the platform.  I'm not sure how you can test the type of a field member under cmake though.

Reply:
hm yea - i just grabbed this from nanomsg & have not tested on net bsd specifically. i'll add a TODO comment and leave it as is for now.